### PR TITLE
Simplify helpers and provides logic & remove RHEL 5 support

### PR DESCRIPTION
**Pull request by tas50**
A collection of misc tweaks:

1) Remove unused helpers
2) Remove Debian 6 support
3) Simplify the service name logic
4) Check for Linux and return before checking obscure Solaris distros
5) Better support Oracle Linux 7+ and SLES

### CODE REVIEW

File: libraries/mysql_service_manager_systemd.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ class MysqlServiceManagerSystemd < MysqlServiceBase
 
     provides :mysql_service_manager, platform: 'fedora'
 
-    provides :mysql_service_manager, platform: %w(redhat centos scientific) do |node| # ~FC005
+    provides :mysql_service_manager, platform: %w(redhat centos scientific oracle) do |node| # ~FC005
       node['platform_version'].to_f >= 7.0
```
Comment (lamont-granquist) [inline libraries/mysql_service_manager_systemd.rb]: all kinds of wrong here, it should ideally use platform_family of `rhel` or `fedora` and then check if systemd is installed but the whole thing should boil down to:

```ruby
  provides :service, os: "linux" do |node|
    Chef::Platform::ServiceHelpers.service_resource_providers.include?(:systemd)
  end
```

I should really add `systemd?` to core from chef-sugar and replace that ugliness.

File: libraries/mysql_service_manager_sysvinit.rb [hunk 2]
```diff
@@ -39,7 +39,7 @@ class MysqlServiceManagerSysvinit < MysqlServiceBase
       end
 
       service mysql_name do
-        provider Chef::Provider::Service::Init::Redhat if node['platform_family'] == 'redhat'
+        provider Chef::Provider::Service::Init::Redhat if %w(redhat suse).include?(node['platform_family'])
```
Comment (lamont-granquist) [inline libraries/mysql_service_manager_sysvinit.rb]: `rhel` is the platform_family, and it should be `%w(rhel fedora suse)`

i'm not sure why you can't rely on the provider resolver picking the correct service provider here and just delete these lines.

i guess conceivably you could force it to use this class on a systemd-supporting host and then wind up with the system provider when you really wanted to force the sysvinit provider -- but can we simply not give any fucks about that use case?

Comment (tas50) [inline libraries/mysql_service_manager_sysvinit.rb]: I believe we did this for sure when Chef 11 did the wrong thing, but will Chef do the right thing when we're trying to force it to use a non-native init system?

File: libraries/mysql_service_manager_systemd.rb (hunk 1 above)
Comment (tas50) [inline libraries/mysql_service_manager_systemd.rb]: Fixed

File: libraries/mysql_service_manager_sysvinit.rb (hunk 2 above)
Comment (tas50) [inline libraries/mysql_service_manager_sysvinit.rb]: Also I fixed the platform family bit. That was entirely broken for a long time clearly.

**Comment by tas50** [comment]
Ping @lamont-granquist and @someara

**Comment by someara** [comment]
:shipit:

**Pull request merged**